frame/session/src/historical/mod.rs

```diff
@@ -31,18 +31,17 @@ use codec::{Encode, Decode};
 use sp_runtime::KeyTypeId;
 use sp_runtime::traits::{Convert, OpaqueKeys};
 use sp_session::{MembershipProof, ValidatorCount};
-use frame_support::{decl_module, decl_storage};
-use frame_support::{Parameter, print};
+use frame_support::{decl_module, decl_storage, Parameter};
+use frame_support::print;
 use sp_trie::{MemoryDB, Trie, TrieMut, Recorder, EMPTY_PREFIX};
 use sp_trie::trie_types::{TrieDBMut, TrieDB};
-use super::{SessionIndex, Module as SessionModule};
+use super::{SessionIndex, Module as SessionModule, ValidatorIdentification};
 
 mod shared;
 pub mod offchain;
 pub mod onchain;
 
-/// Trait necessary for the historical module.
-pub trait Trait: super::Trait {
+pub trait FullValidatorIdentification<AccountId>: ValidatorIdentification<AccountId> {
```

Comment:
Why are these traits split out of `session` and `session:historical`?
The relation between `session::Trait` and `session::historical::Trait` was pretty clear to me earlier - we have something that provides `session` capabilities and then extension, which also stores historical values. Now it feels the split is completely arbitrary and really doesn't provide any value.

Reply:
Splitting out `ValidatorIdentification` is to make `im-online` do not have to rely on `pallet_session::Trait` directly, which was the first attempt to remove `pallet_sesion` from `im-online`. But yeah, `im-online` still hasn't got rid of it. I can continue this direction if we reach some consensus about the purpose of this PR :(.

Splitting out `FullValidatorIdentification` is intended to resolve https://github.com/paritytech/substrate/pull/7127#discussion_r494893281 so that the session users that don't care about the historical feature don't have to touch `FullIndentification`.

Reply:
I'd rather suggest to put everytihng into the `SessionInterface` trait - i.e. to have one trait that encapsulates the entire validator set and all the things required. We should implement this trait in session module (as we do now).